src/api.v2/model/UserAccess.js

```diff
@@ -38,21 +38,51 @@ class UserAccess extends BasicModel {
     return experimentUsers;
   }
 
+  async addToInviteAccess(userEmail, experimentId, role) {
+    return await this.sql
+      .insert({ user_email: userEmail, experiment_id: experimentId, access_role: role })
+      .into(tableNames.INVITE_ACCESS);
+  }
+
+  async registerNewUserAccess(userEmail, userId) {
+    const records = await this.sql
+      .select(['experiment_id', 'access_role'])
+      .from(tableNames.INVITE_ACCESS)
+      .where({ user_email: userEmail });
+
+    const insertPromise = records.map((record) => this.create({
+      user_id: userId,
+      experiment_id: record.experimentId,
+      access_role: record.accessRole,
+    }));
+
+    await Promise.all(insertPromise);
+
+    await this.sql.del().from(tableNames.INVITE_ACCESS).where({ user_email: userEmail });
+  }
+
+  async grantAccess(userId, experimentId, role) {
+    return await this.create(
+      { user_id: userId, experiment_id: experimentId, access_role: role },
+    );
+  }
+
+  async removeAccess(userId, experimentId) {
+    return await this.delete({ experiment_id: experimentId, user_id: userId });
+  }
```

Comment:
this function is not used anywhere in this file -- better delete it in this case